storey/flow.py

```diff
@@ -452,9 +535,21 @@ async def _do(self, event):
             return await self._do_downstream(_termination_obj)
         else:
             element = self._get_event_or_body(event)
-            fn_result = await self._call(element)
+            fn_result = await self._call(element, self._fn)
             await self._do_internal(event, fn_result)
 
+    async def select_outlets(self, event_body) -> Optional[Collection[str]]:
+        if self._outlets_selector:
+            return self._outlets_selector(event_body)
+        else:
+            return super().select_outlets(event_body)
+
+    def _init(self):
+        self._create_name_to_outlet = self._outlets_selector is not None or self._method_is_overridden(
+            "select_outlets", _UnaryFunctionFlow
+        )
+        super()._init()
```

Comment:
We really don't need to make allowance for the user subclassing something like Map and overriding its `select_outlets`. Better give the user one way to do things. In this case, via the `outlets_selector` parameter.
```suggestion
```